sourcemash/frontend/static/js/views/register_modal_view.js

```diff
@@ -0,0 +1,41 @@
+Sourcemash.Views.RegisterModalView = Backbone.View.extend({
+    template: JST['register-modal'],
+
+    events: {
+        'submit #register': 'registerFromModal',
+    },
+
+    registerFromModal: function(e) {
+        e.preventDefault();
+        var formData = JSON.stringify($("#register").serializeObject());
+        $('#register-password').val('');
+        $('#register-password-confirm').val('');
+        var posting = $.ajax({
+                          type: "POST",
+                          url: "/register",
+                          data: formData,
+                          success: this.registerOrShowErrors,
+                          contentType: "application/json"
+                      });
+    },
+
+    registerOrShowErrors: function(data) {
+        var user = data.response.user;
+        if (user) {
+          mixpanel.track("Register", {"User": user});
+          $("#register-modal").closeModal();
+          toast("Check your email for confirmation!", 2000);
+          setTimeout(function(){window.location.replace("/")}, 2000);
+        };
+        var errors = data.response.errors;
```

Comment:
Could we extract this error code to the AJAX catch? We already wrote the logic for rendering errors there, so it can clean up this function a little if we move this there.